### PR TITLE
[Release] Store metrics separately in buildkite artifacts

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

Stores prometheus metrics as a separate file in buildkite artifacts.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Yard1** [comment]
Release test run: https://buildkite.com/ray-project/release-tests-pr/builds/25723

File: release/ray_release/reporter/artifacts.py [hunk 1]
```diff
@@ -27,9 +29,20 @@ def report_result(self, test: Test, result: Result):
...
+            json.dump(result_dict, fp, sort_keys=True, indent=4)
 
         logger.info(
             f"Wrote test config and result to artifacts directory: {self.artifacts_dir}"
         )
+
+        if metrics_dict:
+            metrics_file = os.path.join(self.artifacts_dir, METRICS_RESULT_FILE)
+            with gzip.open(metrics_file, "w") as f:
+                f.write(
```
Comment (krfricke) [inline release/ray_release/reporter/artifacts.py:44]: Let's keep it consistent with the other write instruction

```suggestion
            with gzip.open(metrics_file, "wt") as fp:
                fp.write(
```

File: release/ray_release/reporter/artifacts.py [hunk 2]
```diff
@@ -27,9 +29,20 @@ def report_result(self, test: Test, result: Result):
...
+                    json.dumps(metrics_dict, sort_keys=True, indent=4).encode("utf-8")
+                )
+
+            logger.info(f"Wrote metrics to artifacts directory: {self.artifacts_dir}")
```
Comment (krfricke) [inline release/ray_release/reporter/artifacts.py:48]:
```suggestion
            logger.info(f"Wrote prometheus metrics to artifacts directory: {self.artifacts_dir}")
```

**Review by krfricke — APPROVED** [review]
~Can we run an example test?~ Thanks!

**Pull request merged**